app/controllers/order_detail_stored_files_controller.rb

```diff
@@ -32,7 +32,7 @@ def order_file
 
   # POST /orders/:order_id/order_details/:order_detail_id/upload_order_file
   def upload_order_file
-    @file = @order_detail.stored_files.new(params.require(:stored_file).permit(:file))
+    @file = @order_detail.stored_files.new(params[:stored_file]&.permit(:file))
```

Comment:
Do we need both the hash syntax and the `&.` in order to handle the possible nil? I would have thought require would fail more elegantly than that...

Reply:
`require` raises an error when `:stored_file` doesn't exist in the params. If you don't select a file, the browser doesn't send the param `stored_file[file]`, so we don't get a `stored_file` key.

Another option would be `.new(file: params.dig(:stored_file, :file))`, but that's not really a great pattern for when you need more params.

Or a `rescue`, but that's about it if we want to keep the `require`.

I'm not really sure of a better option.

Reply:
I think it's fine as is then. Thanks for the additional context!